src/main/java/users/model/UsersMap.java

```diff
@@ -20,7 +22,11 @@ public static UsersMap getUsersInstance() {
   }
 
   public void addToUserList(String userName, User user) {
+    if(usersMap.getUser(userName) != null)
     users.put(userName, user);
+    else {
+      throw new UserAlreadyExistsException("This User: "+ userName + "Already Exists");
```

Comment:
I usually don't use string.format, but you should be better than me and use it :)